internal/reconciliations/istio/configuration.go

```diff
@@ -78,6 +83,108 @@ func EvaluateIstioCRChanges(istioCR operatorv1alpha1.Istio, istioTag string) (tr
 	return trigger, nil
 }
 
+func checkComponentsConfigChange(components *operatorv1alpha1.Components, components2 *operatorv1alpha1.Components) IstioCRChange {
```

Comment:
Why does this "check" function return a IstioCRChange and checkK8SConfigChange a boolean? I think it would be easier to use booleans in both cases to keep IstioCRChange out of this level of the functions.